lib/src/merged_tree.rs

```diff
@@ -331,6 +334,24 @@ impl MergedTree {
     }
 }
 
+/// Options for tree/file conflict resolution.
+#[derive(Clone, Debug, Default)]
+pub struct MergeOptions {
+    /// Granularity of hunks when merging files.
+    pub hunk_level: FileMergeHunkLevel,
+}
+
+impl MergeOptions {
+    /// Loads merge options from `settings`.
+    pub fn from_settings(settings: &UserSettings) -> Result<Self, ConfigGetError> {
+        Ok(Self {
+            // Maybe we can add hunk-level=file to disable content merging if
+            // needed. It wouldn't be translated to FileMergeHunkLevel.
+            hunk_level: settings.get("merge.hunk-level")?,
+        })
+    }
+}
+
```

Comment:
(no action requested)

I've been thinking that may we should move `merge_trees()` to a separate file.